riscv-page-tables/src/page_table.rs

```diff
@@ -183,6 +199,13 @@ impl<'a, T: PagingMode> UnusedPte<'a, T> {
         self.pte.lock();
         LockedUnmappedPte::new(self.pte, self.level)
     }
+
+    /// Makes the entry a valid leaf.
+    fn leaf(self) -> LeafPte<'a, T> {
+        self.pte.leaf();
+        self.pte.mark_valid();
+        LeafPte::new(self.pte, self.level)
+    }
```

Comment:
how can we turn an unused PTE into a valid leaf? the PFN is zero at this point, so we need to pass in the PFN to be set.

Reply:
Gotcha, something I had completely missed.

Reply:
Fixed. PTAL